pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/text/TextIndexType.java

```diff
@@ -178,4 +182,24 @@ public TextIndexReader createIndexReader(SegmentDirectory.Reader segmentReader,
       return new LuceneTextIndexReader(metadata.getColumnName(), segmentDir, metadata.getTotalDocs(), indexConfig);
     }
   }
+
+  @Nullable
+  @Override
+  public MutableIndex createMutableIndex(MutableIndexContext context, TextIndexConfig config) {
+    if (config.isDisabled()) {
+      return null;
+    }
+    if (!context.getFieldSpec().isSingleValueField()) {
+      return null;
+    }
+    if (config.getFstType() == FSTType.NATIVE) {
+      return new NativeMutableTextIndex(context.getFieldSpec().getName());
+    } else {
```

Comment:
as return early in the if-branch, can move things out of else branch and save some indents.